src/main/java/com/iota/iri/network/pipeline/TxPipeline.java

```diff
@@ -0,0 +1,209 @@
+    private ValidationStage validationStage;
+    private ReplyStage replyStage;
+    private BroadcastStage broadcastStage;
+    private BatchedHasher batchedHasher;
+    private HashingStage hashingStage;
+
+    private FIFOCache<Long, Hash> recentlySeenBytesCache;
+    private BlockingQueue<ProcessingContext<PreProcessPayload>> preProcessStageQueue = new ArrayBlockingQueue<>(100);
+    private BlockingQueue<ProcessingContext<? extends ValidationPayload>> validationStageQueue = new ArrayBlockingQueue<>(100);
+    private BlockingQueue<ProcessingContext<ReceivedPayload>> receivedStageQueue = new ArrayBlockingQueue<>(100);
+    private BlockingQueue<ProcessingContext<BroadcastPayload>> broadcastStageQueue = new ArrayBlockingQueue<>(100);
+    private BlockingQueue<ProcessingContext<ReplyPayload>> replyStageQueue = new ArrayBlockingQueue<>(100);
+
+    public enum Stage {
+        PRE_PROCESS,
+        HASHING,
+        VALIDATION,
+        REPLY,
+        RECEIVED,
+        BROADCAST,
+        MULTIPLE,
+        ABORT
+    }
+
+    public void init(NeighborRouter neighborRouter, NodeConfig config, TransactionValidator txValidator, Tangle tangle,
+                     SnapshotProvider snapshotProvider, TransactionRequester txRequester,
+                     TipsViewModel tipsViewModel, LatestMilestoneTracker latestMilestoneTracker) {
+        this.recentlySeenBytesCache = new FIFOCache<>(config.getCacheSizeBytes(), config.getpDropCacheEntry());
+        this.preProcessStage = new PreProcessStage(recentlySeenBytesCache);
+        this.replyStage = new ReplyStage(neighborRouter, config, tangle, tipsViewModel, latestMilestoneTracker, recentlySeenBytesCache, txRequester);
+        this.broadcastStage = new BroadcastStage(neighborRouter);
+        this.validationStage = new ValidationStage(txValidator, recentlySeenBytesCache);
+        this.receivedStage = new ReceivedStage(tangle, txValidator, snapshotProvider);
+        this.batchedHasher = BatchedHasherFactory.create(BatchedHasherFactory.Type.BCTCURL81);
+        this.hashingStage = new HashingStage(batchedHasher);
+    }
+
+    public void start() {
+        /*
+            Assembly:
+            pre process stage:
+                -> reply stage if tx already seen
+                -> hashing stage if tx not already seen
+            hashing stage:
+                -> validation stage
+            validation stage:
+                -> received stage if not tx not gotten from a neighbor
+                -> received/reply stage otherwise
+            received stage:
+                -> broadcast stage
+
+            external actors:
+                neighbors:
+                    -> pre process stage
+                broadcastTransactions:
+                    -> hashing stage
+         */
+        stagesThreadPool.submit(batchedHasher);
+        stagesThreadPool.submit(new Thread(() -> {
+            try {
+                while (!NeighborRouter.SHUTDOWN.get()) {
```

Comment:
When did this turn into upper case?
Upper case is only reserved to final constants according to java conventions

change to `isShutdown()` please

Reply:
Reviewed outdated code (?). `TransactionProcessingPipeline` has its own lowercased shutdown field.